server/src/main/java/org/opensearch/indices/recovery/RecoverySourceHandlerFactory.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.indices.recovery;
+
+import org.opensearch.index.shard.IndexShard;
+
+/**
+ * Factory that supplies {@link RecoverySourceHandler}.
+ */
+public class RecoverySourceHandlerFactory {
+
+    public RecoverySourceHandler create(
+        IndexShard shard,
+        RecoveryTargetHandler recoveryTarget,
+        StartRecoveryRequest request,
+        RecoverySettings recoverySettings
+    ) {
+        boolean isReplicaRecoveryWithRemoteTranslog = request.isPrimaryRelocation() == false && shard.isRemoteTranslogEnabled();
```

Comment:
Why are we checking `request.isPrimaryRelocation() == false`?

Reply:
`request.isPrimaryRelocation() == false` is replica recovery. We want to use the RemoteStoreReplicaRecoverySourceHandler only for recovering the replicas. When `request.isPrimaryRelocation() == true`, it is primary-primary recovery for which it would be DefaultRecoverySourceHandler. Primary-primary recovery would involve translogs and lucene operation replay during the peer recovery.